modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/application/HeadlessDiscoveryOpenAPIApplication.java

```diff
@@ -68,168 +33,6 @@
-	}
-
-	@Activate
-	protected void activate(BundleContext bundleContext) {
-		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
-			bundleContext, Application.class, "companyId",
-			new EagerServiceTrackerCustomizer<Application, Application>() {
-
-				@Override
-				public Application addingService(
-					ServiceReference<Application> serviceReference) {
-
-					_populateCompanyIds(serviceReference);
-
-					return bundleContext.getService(serviceReference);
-				}
-
-				@Override
-				public void modifiedService(
-					ServiceReference<Application> serviceReference,
-					Application application) {
-
-					_populateCompanyIds(serviceReference);
-				}
-
-				@Override
-				public void removedService(
-					ServiceReference<Application> serviceReference,
-					Application application) {
-
-					Object osgiJaxRsApplicationBase =
-						serviceReference.getProperty(
-							"osgi.jaxrs.application.base");
-
-					if (osgiJaxRsApplicationBase instanceof String) {
-						_companyIds.remove(osgiJaxRsApplicationBase);
-					}
-
-					bundleContext.ungetService(serviceReference);
-				}
-
-			});
-	}
-
-	@Deactivate
-	protected void deactivate() {
-		_serviceTrackerMap.close();
-	}
-
-	private void _addPaths(
-		String basePath, List<String> paths,
-		ResourceMethodInfoDTO[] resourceMethodInfoDTOS, String serverURL) {
-
-		for (ResourceMethodInfoDTO resourceMethodInfoDTO :
-				resourceMethodInfoDTOS) {
-
-			String path = resourceMethodInfoDTO.path;
-
-			if (path.contains("/openapi")) {
-				String openAPIPath = StringUtil.replace(
-					resourceMethodInfoDTO.path, "{type:json|yaml}", "yaml");
-
-				paths.add(serverURL + basePath + openAPIPath);
-			}
-		}
-	}
-
-	private ApplicationDTO[] _getApplicationDTOs(
-		ApplicationDTO[] applicationDTOS) {
-
-		return ArrayUtil.filter(
-			applicationDTOS,
-			applicationDTO -> {
-				if (_companyIds.containsKey(applicationDTO.base)) {
-					List<String> companyIds = _companyIds.get(
-						applicationDTO.base);
-
-					return companyIds.contains(
-						String.valueOf(CompanyThreadLocal.getCompanyId()));
-				}
-
-				return true;
-			});
-	}
-
-	private void _populateCompanyIds(
-		ServiceReference<Application> serviceReference) {
-
-		Object companyIds = serviceReference.getProperty("companyId");
-		Object osgiJaxRsApplicationBase = serviceReference.getProperty(
-			"osgi.jaxrs.application.base");
-
-		if ((companyIds instanceof List) &&
-			(osgiJaxRsApplicationBase instanceof String)) {
-
-			_companyIds.put(
-				(String)osgiJaxRsApplicationBase, (List<String>)companyIds);
-		}
-	}
-
-	private final Map<String, List<String>> _companyIds = new HashMap<>();
-
-	@Context
-	private HttpServletRequest _httpServletRequest;
-
-	@Reference
-	private JaxrsServiceRuntime _jaxrsServiceRuntime;
-
-	@Reference
-	private Portal _portal;
-
-	private ServiceTrackerMap<String, Application> _serviceTrackerMap;
-
-	@Context
-	private UriInfo _uriInfo;
+	public static final String BASE_PATH = "/openapi";
```

Comment:
Mmmm, letting a constant in an Application class... I don't know if I like it. Should be in a constants class on the `-api` module right? I think it is necessary to create that module for the headless-discovery.

Reply:
Mmmm... I don't know if I prefer to create a new module for a constant, removing the constant and use just the String literal or send it to Brian and in the future, move it to the new module

Reply:
Well, I prefer to have the constant at least here instead of not having it and using [magic strings](https://deviq.com/antipatterns/magic-strings) 😁